app/controllers/application_controller.rb

```diff
@@ -1,47 +1,29 @@
 PROFILER_SESSIONS_FILE = 'used_tags.txt'
 
 class ApplicationController < ActionController::Base
+  include HtmlCleaner
 
   helper :all # include all helpers, all the time
+  helper_method :process_title # Title helpers
 
-  include HtmlCleaner
   before_filter :sanitize_params
+  before_filter :set_current_user
+  before_filter :clear_flash_cookie
 
-  # Authlogic login helpers
-  helper_method :current_user
-  helper_method :current_admin
-  helper_method :logged_in?
-  helper_method :logged_in_as_admin?
+  after_filter :store_location
```

Comment:
Storing redirect data make more sense after the action was performed, since it prevent conflicts with Devise, check all user permissions and we don't get redirect loops, since it won't store it if another redirect is made :)